env0/resource_aws_credentials.go

```diff
@@ -31,13 +31,14 @@ func resourceAwsCredentials() *schema.Resource {
 				ConflictsWith: []string{"access_key_id"},
 			},
 			"external_id": {
-				Type:          schema.TypeString,
-				Description:   "the aws role external id",
-				Optional:      true,
-				Sensitive:     true,
-				ForceNew:      true,
-				ConflictsWith: []string{"access_key_id"},
-				RequiredWith:  []string{"arn"},
+				Type:             schema.TypeString,
+				Description:      "the aws role external id",
+				Optional:         true,
+				Sensitive:        true,
+				ForceNew:         true,
+				ConflictsWith:    []string{"access_key_id"},
+				RequiredWith:     []string{"arn"},
+				ValidateDiagFunc: NewRegexValidator(`^[A-Za-z0-9+=\-.,@:/]{2,1000}$`),
```

Comment:
It took a while to figure out what's wrong. I limited the regex to 1000 (more than 1000 causes a panic in Golang).